api/v1beta1/aerospikecluster_validating_webhook.go

```diff
@@ -1088,30 +1100,42 @@ func validateNsConfUpdate(
 				}
 
 				// storage-engine update not allowed for now
-				storage, ok1 := singleConf["storage-engine"]
-				oldStorage, ok2 := oldSingleConf["storage-engine"]
+				storage, ok1 := singleConf["storage-engine"].(map[string]interface{})
+				oldStorage, ok2 := oldSingleConf["storage-engine"].(map[string]interface{})
 				if ok1 && !ok2 || !ok1 && ok2 {
 					return fmt.Errorf(
 						"storage-engine config cannot be added or removed from existing cluster. Old namespace config %v, new namespace config %v",
 						oldSingleConf, singleConf,
 					)
 				}
-				if ok1 && ok2 && !reflect.DeepEqual(storage, oldStorage) {
+				err := validateStorageConfiguration(oldStorage, storage, singleConf["name"].(string))
```

Comment:
I suggest to rename to validateStorageEngineConfiguration to disambiguate storage term used at multiple places in aerospike conf